apps/parsley/src/utils/filterLogs/index.ts

```diff
@@ -80,18 +80,27 @@ const filterLogs = (options: FilterLogsParams): ProcessedLogLines => {
       for (let idx = 0; idx < logLines.length; idx++) {
         const func = sectionData.functions[funcIndex];
         const isFuncStart = func && idx === func.range.start;
-        const isFuncOpen = sectionState[func?.functionID]?.isOpen ?? false;
+        const isFuncOpen =
+          (sectionState[func?.functionID]?.isOpen ||
+            func?.containsTopLevelCommand) ??
+          false;
 
         const command = sectionData.commands[commandIndex];
         const isCommandStart = command && idx === command.range.start;
 
+        // A function start is detected.
         if (isFuncStart) {
-          // A function start is detected.
-          filteredLines.push({
-            ...func,
-            isOpen: isFuncOpen,
-            rowType: RowType.SectionHeader,
-          });
+          // Don't show the function header if contains a top-level command.
+          if (!func.containsTopLevelCommand) {
```

Comment:
tbh the more I think about it, the more I'm confused about how we have an invisible function that doesn't get rendered. is there not a way to just render a standalone command independent of a function?

Reply:
There are alternative ways to do this but I found them more confusing and would require more refactoring (I can go over them if you like). You can think of the model as so: Instead of promoting a command up to the level of a function, see through the function and access the command directly. The neat thing about this solution is that the casing really only happens at render time (and searching but only for performance reasons).